test/commands/projects/remove.test.js

```diff
@@ -0,0 +1,162 @@
+
+const sinon = require('sinon');
+const { expect, test, constants } = require('@twilio/cli-test');
+const { Config, ConfigData } = require('@twilio/cli-core').services.config;
+const ProjectsRemove = require('../../../src/commands/projects/remove');
+
+describe('commands', () => {
+  describe('projects', () => {
+    describe('remove', () => {
+      const setup = (commandArgs = [], { deleteProject = true, deleteKey = true, removeCred = true } = {}) => test
+        .do(ctx => {
+          ctx.userConfig = new ConfigData();
+          ctx.userConfig.addProject('project1', constants.FAKE_ACCOUNT_SID);
+          ctx.userConfig.addProject('project2', constants.FAKE_ACCOUNT_SID);
+          ctx.userConfig.addProject('project3', constants.FAKE_ACCOUNT_SID);
+          ctx.userConfig.addProject('project4', constants.FAKE_ACCOUNT_SID);
+        })
+        .twilioCliEnv(Config)
+        .twilioCreateCommand(ProjectsRemove, commandArgs)
+        .stdout()
+        .stderr()
+        .do(ctx => {
+          const fakePrompt = sinon.stub();
+          fakePrompt
+            .onFirstCall()
+            .resolves({
+              affirmative: deleteProject
+            })
+            .onSecondCall()
+            .resolves({
+              affirmative: deleteKey
+            });
+          ctx.testCmd.inquirer.prompt = fakePrompt;
+        })
+        .do(ctx => {
+          ctx.testCmd.secureStorage.removeCredentials = () => {
+            return removeCred;
+          };
+        })
+        .do(ctx => ctx.testCmd.run());
+
+      const oneSetup = (commandArgs = [], { deleteProject = true, deleteKey = true, removeCred = true } = {}) => test
+        .do(ctx => {
+          ctx.userConfig = new ConfigData();
+          ctx.userConfig.addProject('project1', constants.FAKE_ACCOUNT_SID);
+        })
+        .twilioCliEnv(Config)
+        .twilioCreateCommand(ProjectsRemove, commandArgs)
+        .stdout()
+        .stderr()
+        .do(ctx => {
+          const fakePrompt = sinon.stub();
+          fakePrompt
+            .onFirstCall()
+            .resolves({
+              affirmative: deleteProject
+            })
+            .onSecondCall()
+            .resolves({
+              affirmative: deleteKey
+            });
+          ctx.testCmd.inquirer.prompt = fakePrompt;
+        })
+        .do(ctx => {
+          ctx.testCmd.secureStorage.removeCredentials = () => {
+            return removeCred;
+          };
+        })
+        .nock('https://api.twilio.com', api => {
+          api.delete(`/2010-04-01/Accounts/${constants.FAKE_ACCOUNT_SID}/Keys/${constants.FAKE_API_KEY}.json`).reply(200, {
+            sid: constants.FAKE_API_KEY,
+            secret: constants.FAKE_API_SECRET
+          });
+        })
+        .do(ctx => ctx.testCmd.run());
+
+      const setupNock = (commandArgs = [], { deleteProject = true, deleteKey = true, removeCred = true } = {}) => test
```

Comment:
These setup functions are almost the same. Can it be refactored to move out the common pieces?